XA Transaction recovery support is enabled even thoug transaction-xa-enabled is false

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

Trying to run Keycloak 25 in Kubernetes with read-only root filesystem will fail with
```
com.arjuna.ats.arjuna.exceptions.ObjectStoreException: ARJUNA012225: FileSystemStore::setupStore - cannot access root of object store: /opt/keycloak/bin/../data/transaction-logs/ShadowNoFileLockStore/defaultStore/
```

From [upgrading guide](https://www.keycloak.org/docs/25.0.0/upgrading/#xa-transaction-changes)
> The option transaction-xa-enabled will default to false
> XA Transaction recovery support is enabled by default if transaction-xa-enabled is true

I came to the conclusion, that recovery should be disabled by default, because XA is disabled. But it is not the case as we can see from log.

### Version

25.0.6

### Regression

- [ ] The issue is a regression

### Expected behavior

Start successfully

### Actual behavior

Fails to start

### How to Reproduce?

1. Build custom optimized image
```
FROM quay.io/keycloak/keycloak:${VERSION}

ENV KC_CACHE=local
ENV KC_DB=postgres
ENV KC_HEALTH_ENABLED=true
ENV KC_METRICS_ENABLED=true

RUN /opt/keycloak/bin/kc.sh build
```
2. Try to run
```
services:
  keycloak:
    image: registry/keycloak:std-pg-25.0.6
    container_name: keycloak
    command: start
    environment:
      KC_DB: postgres
      KC_DB_URL: jdbc:postgresql://postgres:5432/keycloak
      KC_DB_USERNAME: keycloak
      KC_DB_PASSWORD: password
      KC_HOSTNAME: localhost
      KC_HTTP_ENABLED: true
      KEYCLOAK_ADMIN: admin
      KEYCLOAK_ADMIN_PASSWORD: admin
    depends_on:
      - postgres
    ports:
      - 8080:8080
    volumes:
      - keycloak_tmp:/tmp
    read_only: true

  postgres:
    image: postgres:16
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: keycloak
      POSTGRES_USER: keycloak
      POSTGRES_PASSWORD: password

volumes:
  postgres_data:
  keycloak_tmp:
```
3. Get error

```
keycloak    | 2024-09-26 13:12:35,168 WARN  [com.arjuna.ats.common] (main) ARJUNA048006: cannot create new instance of com.arjuna.ats.internal.arjuna.objectstore.ShadowNoFileLockStore: java.lang.reflect.InvocationTargetException
keycloak    | Caused by: com.arjuna.ats.arjuna.exceptions.ObjectStoreException: ARJUNA012225: FileSystemStore::setupStore - cannot access root of object store: /opt/keycloak/bin/../data/transaction-logs/ShadowNoFileLockStore/defaultStore/
keycloak    | 2024-09-26 13:12:35,175 INFO  [com.arjuna.ats.jbossatx] (main) ARJUNA032014: Stopping transaction recovery manager
```

### Anything else?

Works as expected if build/run with `QUARKUS_TRANSACTION_MANAGER_ENABLE_RECOVERY: false`.
There is also a hack to mount `emptyDir` volume to `/opt/keycloak/data`.

--- a/docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc
+++ b/docs/documentation/upgrading/topics/changes/changes-25_0_0.adoc
@@ -250,7 +250,7 @@ The module `org.keycloak:keycloak-model-legacy` module was deprecated in a previ
 = XA Transaction Changes
 
 * The option `transaction-xa-enabled` will default to false, rather than true. If you want XA transaction support you will now need to explicitly set this option to true.
-* XA Transaction recovery support is enabled by default if `transaction-xa-enabled` is true. Transaction logs will be stored at KEYCLOAK_HOME/data/transaction-logs.
+* XA Transaction recovery support is enabled by default. Transaction logs will be stored at KEYCLOAK_HOME/data/transaction-logs.
 
 = Removed offline session preloading
 

